### ISSUE TITLE
What would happen if transaction id is null and merchant tries to open the dispute details?

### ISSUE THREAD

**Issue by shendy-a8c** [Automattic/woocommerce-payments#8453]
### Describe the bug

We just found out that it's possible `balance_transaction` from dispute object is null (read p1711102875028819/1711018991.005889-slack-C02BW3Z8SHK), for example transactions paid using Klarna.

Since the dispute link requires transaction id ([ref](https://github.com/Automattic/woocommerce-payments/blob/bc8ae14f29921f43f105a2240d21fe5c23ee932a/client/disputes/redirect-to-transaction-details/index.tsx#L33C34-L33C53)), dispute form Klarna transactions might be breaking as we speak.

### To Reproduce

Not sure how but, make a purchase using Klarna and somehow have that transaction disputed.

**Comment by haszari** [comment]
Is this at all related to our other Klarna dispute issue? 

- #7885

@nagpai (porter) - could you please read through the linked slack discussion and update the description with more info? Feel free to timebox to 10-15 minutes. Info such as

- When the bug happens (data is missing/bad), how is the UI or merchant experience broken?
- How does the bug occur or how to reproduce (this might be tricky – whatever info we have).

FYI @brucealdridge  as you did some work investigating #7885. Please comment / edit description if you have any insights, e.g. are these issues related at all?

**Comment by haszari** [comment]
Also potentially related to:

- #4981

Leaving as `No status` and no priority until we have further info. Likely edge case/low priority, but need more info.

**Comment by brucealdridge** [comment]
I've hit this issue independently when testing inquiries on the Server. 

I think what is happening is the code expects all disputes to have a `balance_transaction`. The [Stripe docs](https://docs.stripe.com/api/disputes/object#dispute_object-balance_transactions) mention that there will be 0, 1 or 2 objects.

```
{
    "id": "dp_1Owy3hQtaq9RariTUd3aG4mN",
    "object": "dispute",
    "amount": 16800,
    "balance_transaction": "txn_1Owy6iQtab9RariTngf8M0CS",
    "balance_transactions": [
        {
            "id": "txn_1Owy3iQtab9RariTnaf8M0CS",
            "object": "balance_transaction",
            "amount": -16800,
            "available_on": 1711411200,
            "created": 1711075337,
            "currency": "usd",
            "description": "Chargeback withdrawal for ch_3Owy3hQtab9RaviT1oBx59Rx",
            "exchange_rate": null,
            "fee": 1500,
            "fee_details": [
                {
                    "amount": 1500,
                    "application": null,
                    "currency": "usd",
                    "description": "Dispute fee",
                    "type": "stripe_fee"
                }
            ],
            "net": -18300,
            "reporting_category": "dispute",
            "source": "dp_1Owy3hQtaq9RariTUd3aG4mN",
            "status": "pending",
            "type": "adjustment"
        }
    ],
    ...
}
```

The `balance_transaction` contains a reference to a transaction where Stripe has held the disputed funds. If the merchant wins the dispute then additional transactions will be added to the `balance_transactions` array.

For inquiries, there are no fees or funds held. So there is no associated `balance_transaction`. Based on what I have found recently, Klarna makes up a large majority of all inquiries hence why we are seeing the relationship here.

**Comment by haszari** [comment]
Thank you @brucealdridge ! So this is essentially technical debt, our code does not handle a case that it should. Follow up questions:

1. Do you think this is `medium` priority or is it higher? 
2. Is there a relatively straightforward fix – i.e. could we patch our logic without risk, even if we don't have a good way to test or reproduce the buggy behaviour?

**Comment by brucealdridge** [comment]
I'm going to close this issue.

This issue is for a hypothetical bug that does not exist. The code that @shendy-a8c called out was released in WooPayments 6.6, if there were problems we should have heard about it by now.

As noted above, Inquiries have `balance_transaction` set to null. Looking at the dispute listing page, neither disputes or inquiries have a `transaction_id` in the URL. When clicking through from the transactions page, both have a `transaction_id` in the URL.  So the URL would work regardless whether a `transaction_id` is present

The same Inquiry, clicking through from _Transactions_ and _Disputes_
![Screenshot 2024-03-25 at 2 48 21 PM](https://github.com/Automattic/woocommerce-payments/assets/57298/2a12a7bd-5b88-4241-9090-220393a5caa3)
![Screenshot 2024-03-25 at 2 48 07 PM](https://github.com/Automattic/woocommerce-payments/assets/57298/f5feffe9-c83f-4d3a-933a-1227a9a089a7)
